File: weblayer/browser/java/org/chromium/weblayer_private/WebViewCompatibilityHelperImpl.java
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.weblayer_private;

import org.chromium.base.annotations.JNINamespace;
import org.chromium.base.annotations.MainDex;
import org.chromium.base.annotations.NativeMethods;

/**
 * Minimal native interface to allow registering all natives if WebView compatibility is needed.
 */
@MainDex
@JNINamespace("weblayer")
public final class WebViewCompatibilityHelperImpl {
    public static void registerJni() {
        WebViewCompatibilityHelperImplJni.get().registerJni();
    }

    @NativeMethods
    interface Natives {
        void registerJni();
    }
}

File: weblayer/browser/web_view_compatibility_helper_impl.cc
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "weblayer/browser/java/jni/WebViewCompatibilityHelperImpl_jni.h"

#if defined(WEBLAYER_MANUAL_JNI_REGISTRATION)
#include "base/android/library_loader/library_loader_hooks.h"  // nogncheck
#include "weblayer/browser/java/weblayer_jni_registration.h"   // nogncheck
#endif

namespace weblayer {
namespace {
#if defined(WEBLAYER_MANUAL_JNI_REGISTRATION)
void RegisterNonMainDexNativesHook() {
  RegisterNonMainDexNatives(base::android::AttachCurrentThread());
}
#endif
}  // namespace

void JNI_WebViewCompatibilityHelperImpl_RegisterJni(JNIEnv* env) {
#if defined(WEBLAYER_MANUAL_JNI_REGISTRATION)
  RegisterMainDexNatives(base::android::AttachCurrentThread());
  base::android::SetNonMainDexJniRegistrationHook(
      RegisterNonMainDexNativesHook);
#endif
}

}  // namespace weblayer
